granulate_utils/linux/cgroups/cgroup.py

```diff
@@ -78,7 +101,94 @@ def find_v2_hierarchy(resolve_host_root_links: bool = True) -> Optional[str]:
     return path
 
 
-def get_cgroup_mount(controller: str, resolve_host_root_links: bool = True) -> Optional[str]:
+CGROUP_PROCS_FILE = "cgroup.procs"
+
+
+class CgroupCore:
+    path: Path
+
+    def __init__(self, path: Path):
+        self.path = path
+
+    def _get_parent_cgroup(self, cgroup_name: str) -> Optional[Path]:
+        if cgroup_name in self.path.parts:
+            return Path(self.path.as_posix().split(cgroup_name)[0]) / cgroup_name
+        return None
+
+    def _get_cgroup_in_hierarchy(self, cgroup_name: str) -> Optional[Path]:
+        if self.has_parent_cgroup(cgroup_name):
+            new_cgroup_path = self._get_parent_cgroup(cgroup_name)
+        else:
+            new_cgroup_path = self.path / cgroup_name
+            new_cgroup_path.mkdir(exist_ok=True)
+
+        return new_cgroup_path
+
+    def has_parent_cgroup(self, cgroup_name: str) -> bool:
+        """
+        Check if cgroup has the given cgroup_name in its hierarchy
+        :param cgroup_name: the cgroup name
+        :return: bool indicating if cgroup is in hierarchy
+        """
+        return self._get_parent_cgroup(cgroup_name) is not None
+
+    def get_pids_in_cgroup(self) -> set[int]:
+        return {int(proc) for proc in self.read_from_interface_file(CGROUP_PROCS_FILE).split()}
+
+    def assign_process_to_cgroup(self, pid: int = 0) -> None:
+        self.write_to_interface_file(CGROUP_PROCS_FILE, str(pid))
```

Comment:
This is consistent behavior of assigning the PROCESS in both v1 and v2, right? Document for clarity with v1 differences and `tasks` file.